File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nums1=list(nums)
        nums1.sort()
        l=0
        r=len(nums1)-1
        while l<r:
            if nums1[l]+nums1[r]==target:
                k=nums1[l]
                m=nums1[r]  
                x=nums.index(k)
                if(k==m):
                    return [x,list(nums[x+1:]).index(m)+x+1]
                return [x,nums.index(m)]
            elif nums1[l]+nums1[r]>target:
                r=r-1
            else:
                l=l+1

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<pair<int,int>> v(nums.size());
        for(int i=0;i<nums.size();i++){
            v[i]={nums[i],i};
        }
        sort(v.begin(),v.end());
        int l=0,r=nums.size()-1;
        vector<int> ans={-1,-1};
        while(l<r){
            if(v[l].first+v[r].first>target){
                r--;
            }
            else if(v[l].first+v[r].first==target){
                ans={v[l].second,v[r].second};
                break;
            }
            else{
                l++;
            }
        }
        
        return ans;
    }
};